### DISCUSSION TITLE
It should be easier to find the file you need to make a ipxe usb boot drive !

### DISCUSSION THREAD

**Discussion by shodanx2** [ipxe/ipxe#757] [General]
I've been searching for a while now and I just can't find a file to write with rufus or balena etcher on my usb drive to make it a ipxe usb boot drive

**Comment by NiKiZe** [comment]
If you want a pcbios version you need to write the raw image. If you want efi then you need ipxe.efi as bootx64.efi. Rufus is likely the wrong tool for the job.

Documentation for what can be built https://ipxe.org/appnote/buildtargets

Additionally there is documentation on how to build a mixed pcbios+efi image, making it easier to find that would be great.

**Reply by shodanx2** [reply]
Please put yourself in the shoes of a novice computer repair shop tech.
They likely don't have a working compiler toolchain nor a linux computer to build it on
I've been hacking at this PXE stuff on my spare time and it's been a very long road to walk

There really needs to be an easily accessible file from the official pxe website that you can just download and drag and drop to your favorite usb stick writer and then boot it.

That's what I wanted to do in the beginning but couldn't really find how, or what it would boot to if I even did. 

Now I realize, I could have saved myself a lot of grief if I had such a usb ipxe test stick from the beginning because I just lost so much time dealing with quirky bioses PXE implementations

I think the ideal ipxe usb stick should be,
[] a single file
[] writable by your favorite usb stick writer (rufus or balena etcher, maybe others)
[] has all the files for either legacy bios and uefi bios
[] should come with easy to use default configuration that give user options to manually try things even if the internet is unaccessible
[] boot.cfg and menu.ipxe easily accessible on a fat32/exfat partition (or maybe even ntfs)

That would have been great help

Of course by now I have many linux boxes, with multiple pxe servers, ipxe-buildweb, netboot.xyz, windows server WDS and just the whole nine yards so it's too late for me. 

Just trying to save some grief to the people walking down this path in the future. 
I could be so much easier

In fact, this file writes with rufus and boots on, at least uefi x86_64
https://boot.netboot.xyz/ipxe/netboot.xyz.img

And this iso writes on rufus and maybe boots on some system, but not mine
https://boot.netboot.xyz/ipxe/netboot.xyz.iso

**Reply by stappersg** [reply]
> Just trying to save some grief

_patches welcome_

**Reply by NiKiZe** [reply]
That is where I come from myself, and I have seen plenty of stories where people look for "simple solutions" which is sometimes given, and then we get back "it doesn't work, iPXE sucks"

What you are asking might not be possible, at least not from our side.
Remember, iPXE is a boot loader. Whatever modifier you are using needs to be aware of this. You have ipxe.usb from boot.ipxe.org, which you copy with DD or similar tool, does Rufus have a dd mode?

**Reply by stappersg** [reply]
Because no answer yet on _does Rufus have a dd mode?_ repeating the preamble text, this time formatted.

Remember, _iPXE_ is a boot loader. Whatever modifier you are using needs to be aware of this. You have `ipxe.usb` from http://boot.ipxe.org which you copy with `dd` or similar tool.

I hope that this comment triggers either _Yes, Rufus has such mode_ or _I don't know `dd` and `Rufus` good enough to tell if it has_.  So that the discussion continues, brings mutual understanding.

**Comment by stappersg** [comment]
(After writing https://github.com/ipxe/ipxe/discussions/757#discussioncomment-3785333 ) And I think this discussion should be retitled from _"It should be easier to find the file you need to make a ipxe usb boot drive!"_ into _"It should be easier to find the tool you need to make a iPXE USB boot drive"_

**Reply by shodanx2** [reply]
I think it really should be a file.
In the time I've had this week, I've picked up a file called netboot.xyz.img, wrote that to usb and it worked. Theorically I could almost using without knowing what a pxe is. Maybe sending the users there would be a good first step. 

Well, except that assumes your router is setup for pxe and serves local-vars.ipxe

I also got ipxe-webbuilder going. A copy of that should be hosted on ipxe.org and easy to find from the download page. Maybe even several buildconfig links for the most common applications.

Once you have it working, it gets so much easier but finding where to begin is so hard !

I feel like every body walking this path has to deal with the same obstacles over and over again. 
And the obstacles are all so minor after you've found the solution

![image](https://user-images.githubusercontent.com/10621885/193568320-b4b8f8f2-abbc-40ac-a426-e6bd613f27b6.png)

**Reply by NiKiZe** [reply]
So again https://boot.ipxe.org/ipxe.usb !

**Reply by stappersg** [reply]
> So again https://boot.ipxe.org/ipxe.usb !

Yes, however it is **not** _clickable_ from https://ipxe.org/download ( https://boot.ipxe.org/ipxe.iso is )

The original _I just can't find a file to write with rufus or balena etcher on my usb drive to make it a ipxe usb boot drive_ seems to be _I just can't find a file to write to my usb drive to make it a ipxe usb boot drive_, so leaving out the tool part. Yeah, that is on hindsight, we are meanwhile beyond the confusion slash misunderstanding on tool to use.

And for what it is worth, shodanx2 does remind me of my younger self. Probably reminds us all of the moment realizing that simple looking things are not always simple.

**Reply by stappersg** [reply]
> > So again https://boot.ipxe.org/ipxe.usb !
> 
> Yes, however it is **not** _clickable_ from https://ipxe.org/download ( https://boot.ipxe.org/ipxe.iso is )

Now _work in progress_ at #765
